lib/board.rb

```diff
@@ -0,0 +1,48 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(pos)
+    self.cells[(pos.to_i) - 1]
+  end
+
+  def full?
+    self.cells.all? {|cell| cell == "X" || cell == "O"}
+  end
+
+  def turn_count
+    self.cells.select {|cell| cell == "X" || cell == "O"}.size
+  end
+
+  def taken?(pos)
+    position(pos) != " " && position(pos) != nil ? true : false
+  end
+
+  def valid_move?(pos)
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
pos.to_i.between?(1,9) && !taken(pos)
```